galley/pkg/runtime/resource/event.go

```diff
@@ -0,0 +1,72 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package resource
+
+import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+// EventKind is the type of an event.
+type EventKind int
+
+const (
+	// None is a sentinel value. Should not be used.
+	None EventKind = iota
```

Comment:
Why do we need the sentinel `None` value?

Reply:
It is useful to have sentinels that are defaults. It avoids accidentally conveying wrong information. An Event that is uninitialized will have "None" as its kind, which would be very easy to catch.